// @aws-cdk/cx-api/lib/legacy-moved.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadBalancerIpAddressType = exports.VpcSubnetGroupType = exports.SynthesisMessageLevel = exports.UNKNOWN_REGION = exports.UNKNOWN_ACCOUNT = exports.SSMPARAM_NO_INVALIDATE = exports.PROVIDER_ERROR_KEY = exports.PATH_METADATA_KEY = exports.ENDPOINT_SERVICE_AVAILABILITY_ZONE_PROVIDER = exports.AVAILABILITY_ZONE_FALLBACK_CONTEXT_KEY = exports.ASSET_RESOURCE_METADATA_PROPERTY_KEY = exports.ASSET_RESOURCE_METADATA_PATH_KEY = exports.ASSET_RESOURCE_METADATA_IS_BUNDLED_KEY = exports.ASSET_RESOURCE_METADATA_ENABLED_CONTEXT = exports.ASSET_RESOURCE_METADATA_DOCKERFILE_PATH_KEY = exports.ASSET_RESOURCE_METADATA_DOCKER_OUTPUTS_KEY = exports.ASSET_RESOURCE_METADATA_DOCKER_CACHE_TO_KEY = exports.ASSET_RESOURCE_METADATA_DOCKER_CACHE_FROM_KEY = exports.ASSET_RESOURCE_METADATA_DOCKER_CACHE_DISABLED_KEY = exports.ASSET_RESOURCE_METADATA_DOCKER_BUILD_TARGET_KEY = exports.ASSET_RESOURCE_METADATA_DOCKER_BUILD_SSH_KEY = exports.ASSET_RESOURCE_METADATA_DOCKER_BUILD_SECRETS_KEY = exports.ASSET_RESOURCE_METADATA_DOCKER_BUILD_ARGS_KEY = exports.ASSET_PREFIX_SEPARATOR = void 0;
exports._convertCloudAssembly = _convertCloudAssembly;
exports._convertCloudAssemblyBuilder = _convertCloudAssemblyBuilder;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
/**
 * Re-exports of members and types that historically used to live here, but have been moved to a more apt package.
 *
 * Re-exported here for backwards compatibility.
 *
 * To satisfy the (jsii) use of these types in the public API of this package, we do the following:
 *
 * - Re-declare the types that we use in this assembly here (`declare class ...`).
 * - Copy the implementation from the upstream library.
 *
 * That way we satisfy the jsii and backwards compatibility checker, while keeping the implementation
 * centralized in a single place.
 */
const cloud_assembly_api_1 = require("@aws-cdk/cloud-assembly-api");
Object.defineProperty(exports, "ASSET_PREFIX_SEPARATOR", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_PREFIX_SEPARATOR; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_DOCKER_BUILD_ARGS_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_DOCKER_BUILD_ARGS_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_DOCKER_BUILD_SECRETS_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_DOCKER_BUILD_SECRETS_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_DOCKER_BUILD_SSH_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_DOCKER_BUILD_SSH_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_DOCKER_BUILD_TARGET_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_DOCKER_BUILD_TARGET_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_DOCKER_CACHE_DISABLED_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_DOCKER_CACHE_DISABLED_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_DOCKER_CACHE_FROM_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_DOCKER_CACHE_FROM_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_DOCKER_CACHE_TO_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_DOCKER_CACHE_TO_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_DOCKER_OUTPUTS_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_DOCKER_OUTPUTS_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_DOCKERFILE_PATH_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_DOCKERFILE_PATH_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_ENABLED_CONTEXT", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_ENABLED_CONTEXT; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_IS_BUNDLED_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_IS_BUNDLED_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_PATH_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_PATH_KEY; } });
Object.defineProperty(exports, "ASSET_RESOURCE_METADATA_PROPERTY_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.ASSET_RESOURCE_METADATA_PROPERTY_KEY; } });
Object.defineProperty(exports, "AVAILABILITY_ZONE_FALLBACK_CONTEXT_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.AVAILABILITY_ZONE_FALLBACK_CONTEXT_KEY; } });
Object.defineProperty(exports, "ENDPOINT_SERVICE_AVAILABILITY_ZONE_PROVIDER", { enumerable: true, get: function () { return cloud_assembly_api_1.ENDPOINT_SERVICE_AVAILABILITY_ZONE_PROVIDER; } });
Object.defineProperty(exports, "PATH_METADATA_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.PATH_METADATA_KEY; } });
Object.defineProperty(exports, "PROVIDER_ERROR_KEY", { enumerable: true, get: function () { return cloud_assembly_api_1.PROVIDER_ERROR_KEY; } });
Object.defineProperty(exports, "SSMPARAM_NO_INVALIDATE", { enumerable: true, get: function () { return cloud_assembly_api_1.SSMPARAM_NO_INVALIDATE; } });
Object.defineProperty(exports, "UNKNOWN_ACCOUNT", { enumerable: true, get: function () { return cloud_assembly_api_1.UNKNOWN_ACCOUNT; } });
Object.defineProperty(exports, "UNKNOWN_REGION", { enumerable: true, get: function () { return cloud_assembly_api_1.UNKNOWN_REGION; } });
/**
 * @deprecated The official definition of this type has moved to `@aws-cdk/cloud-assembly-api`.
 */
var SynthesisMessageLevel;
(function (SynthesisMessageLevel) {
    SynthesisMessageLevel["INFO"] = "info";
    SynthesisMessageLevel["WARNING"] = "warning";
    SynthesisMessageLevel["ERROR"] = "error";
})(SynthesisMessageLevel || (exports.SynthesisMessageLevel = SynthesisMessageLevel = {}));
exports.CloudAssembly = cloud_assembly_api_1.CloudAssembly;
// Make these classes equal to the upstream versions
exports.CloudArtifact = cloud_assembly_api_1.CloudArtifact;
exports.CloudFormationStackArtifact = cloud_assembly_api_1.CloudFormationStackArtifact;
exports.AssetManifestArtifact = cloud_assembly_api_1.AssetManifestArtifact;
exports.NestedCloudAssemblyArtifact = cloud_assembly_api_1.NestedCloudAssemblyArtifact;
exports.TreeCloudArtifact = cloud_assembly_api_1.TreeCloudArtifact;
exports.CloudAssemblyBuilder = cloud_assembly_api_1.CloudAssemblyBuilder;
function _convertCloudAssembly(x) {
    return x;
}
function _convertCloudAssemblyBuilder(x) {
    return x;
}
exports.EnvironmentPlaceholders = cloud_assembly_api_1.EnvironmentPlaceholders;
exports.EnvironmentUtils = cloud_assembly_api_1.EnvironmentUtils;
/**
 * The type of subnet group.
 * Same as SubnetType in the aws-cdk-lib/aws-ec2 package,
 * but we can't use that because of cyclical dependencies.
 *
 * @deprecated The definition of this type has moved to `@aws-cdk/cloud-assembly-api`.
 */
var VpcSubnetGroupType;
(function (VpcSubnetGroupType) {
    /** Public subnet group type. */
    VpcSubnetGroupType["PUBLIC"] = "Public";
    /** Private subnet group type. */
    VpcSubnetGroupType["PRIVATE"] = "Private";
    /** Isolated subnet group type. */
    VpcSubnetGroupType["ISOLATED"] = "Isolated";
})(VpcSubnetGroupType || (exports.VpcSubnetGroupType = VpcSubnetGroupType = {}));
/**
 * Load balancer ip address type.
 *
 * @deprecated The definition of this type has moved to `@aws-cdk/cloud-assembly-api`.
 */
var LoadBalancerIpAddressType;
(function (LoadBalancerIpAddressType) {
    /**
     * IPV4 ip address
     */
    LoadBalancerIpAddressType["IPV4"] = "ipv4";
    /**
     * Dual stack address
     */
    LoadBalancerIpAddressType["DUAL_STACK"] = "dualstack";
    /**
     * IPv6 only public addresses, with private IPv4 and IPv6 addresses
     */
    LoadBalancerIpAddressType["DUAL_STACK_WITHOUT_PUBLIC_IPV4"] = "dualstack-without-public-ipv4";
})(LoadBalancerIpAddressType || (exports.LoadBalancerIpAddressType = LoadBalancerIpAddressType = {}));
